src/weatherdata/ipm.py

```diff
@@ -54,18 +61,23 @@ def station_ids(self):
         coord = dict()
 
         for names in values:
-            if 'features' in values[names]:
+            if 'features' in values[names] and values[names]['features']!=[]:
                 station_ids[names]=[values[names]['features'][item]['properties'] for item in range(len(values[names]['features']))]
                 coord[names]=[values[names]['features'][item]['geometry'] for item in range(len(values[names]['features']))]
             else:
                 station_ids[names]= 'no stations for this ressources'
                 coord[names]= 'no stations for this ressources'
+        
+        if station_ids[self.name]=='no stations for this ressources':
```

Comment:
Warning: do not put the message in the code.
Perhaps, one solution would be to manage a code associated with the message in a separate module?